### DISCUSSION TITLE
QDX HF Transceivers controlled by a Raspberry Pi's (Reticulum)

### DISCUSSION THREAD

**Discussion by jacqueszsx** [markqvist/Reticulum#261] [Help & Questions]
Any thoughts on using QDX HF Transceivers controlled by a Raspberry Pi's utilizing the unsigned.io software stack. Also using "Sideband? or .." it in a similar way as example included end of message (https://unsigned.io/software/index.html) "HF mesh network for messaging"

QRP Labs  digi-mode HF transceiver QDX 
https://qrp-labs.com/qdx.html
https://qrp-labs.com/images/qdx/fdim2022.pdf
https://qrp-labs.com/images/qdx/qdxseminar.pdf

An example of QDX used with Android Phone (HF Pager App)
https://groups.io/g/QRPLabs/topic/97477592#100184
That's it, other than the 40M Hamstick on the rear bumper and a qrp tuner, which could be replaced with a cheap manual tuner. That's an On The Go (OTG) cable on the cell phone.
Swapped messages with the home station as far as 354 miles away and no skip hole, either. Tested every few miles out to 50 miles and more sparsely the rest of the way. HF Pager http://www.dxsoft.com, for bigger files (I)FSKchat and TrueTTY (PS2x5 and PS33x5 modes).

**Comment by simplyequipped** [comment]
Reticulum over HF radio has been a goal of mine as well, specifically using the QDX and Raspberry Pi. My initial plan for integrating Sideband has been to use the Pi's wifi in hotspot mode and connect to it with a smartphone running Sideband. I would also like to experiment more with connecting RNode's to a smartphone and the PI, to replace the wifi link and increase the range of the local portion of the network.

Here are a few projects I have been working on related to Reticulum over HF in case they are useful to you. Feel free to provide feedback on any of these projects. They can all be improved.

[pyjs8call](https://github.com/simplyequipped/pyjs8call)
This isn't Reticulum specific at this point because its scope has grown quite a bit since inception, but the initial concept was to make JS8Call accessible for integration with Reticulum by exposing and expanding the JS8Call API as well as automating as much of the JS8Call networking features as possible so the application can run "behind the scenes". The ultimate goal is to allow the use of JS8Call as a modem interface that is invisible to the end user (beyond initial configuration).

[portal](https://github.com/simplyequipped/portal)
This project needs updated to the latest release of pyjs8call, but its goal is to provide a simple web based UI for basic text messaging style messaging from a smartphone or computer by connecting to a Raspberry Pi running JS8Call, pyjs8call, and the web app server. This project does not utilize Reticulum since Sideband is a much better tool for that. This project has been functional as of a few months ago, but needs updated to be functional again. This was developed as an interim solution while Reticulum was improved to support extremely low data rates (i.e. JS8Call, which is ~10 baud on fast speed).

[fskmodem](https://github.com/simplyequipped/fskmodem)
[tcpkissserver](https://github.com/simplyequipped/tcpkissserver)
These two projects were developed together with the express purpose of 300 baud packet integration with Reticulum on a Raspberry Pi. I need to revisit these projects myself because I haven't touched them in a while. 300 baud packet isn't the most robust (especially compared to JS8Call) but this setup would give the best data rate over HF, as long as you weren't trying to reach out too far and you had a good antenna to maximize efficiency for something like NVIS. It sounds like you have already done some interesting testing with an NVIS setup.

[qdx](https://github.com/simplyequipped/qdx)
This project was also developed with the express purpose of integrating the QDX with Reticulum on a Raspberry Pi. I need to revisit this project as well, and it is possible that this is not in a working state at this point, but I know this was working in the past.

Also, see discussion #11 which is somewhat related to this discussion.

**Reply by grevemes** [reply]
I was working on this over the weekend and got the QDX working with a Pi as well as a regular workstation.  I used Direwolf/Rigctl as the interface between the computer and the modem. 

Now I need to make it more portable and take it farther than the other end of the workbench.... :)

**Reply by markqvist** [reply]
Wow, that is very cool to hear. If this turns out to be feasible, maybe we could add some sort of interface glue to make it easier. It would be pretty simple to run the CAT control of the radio directly by a Reticulum interface at least, and I'm thinking it might also be possible to automate the process of spawning the softmodem process.
